site/content/how-to/monitoring/prometheus.md

```diff
@@ -11,13 +11,85 @@ docs: "DOCS-1418"
 ## Overview
 
 
-NGINX Gateway Fabric metrics are displayed in [Prometheus](https://prometheus.io/) format, simplifying monitoring. You can track NGINX and controller-runtime metrics through a metrics server orchestrated by the controller-runtime package. These metrics are enabled by default and can be accessed on HTTP port `9113`.
-
+NGINX Gateway Fabric metrics are displayed in [Prometheus](https://prometheus.io/) format, simplifying monitoring. These metrics are served via a metrics server orchestrated by the controller-runtime package on HTTP port `9113`. When installed, Prometheus automatically scrapes this port and collects metrics. [Grafana](https://grafana.com/) can be used for rich visualization of these metrics.
 
 {{<call-out "important" "Security note for metrics">}}
 Metrics are served over HTTP by default. Enabling HTTPS will secure the metrics endpoint with a self-signed certificate. When using HTTPS, adjust the Prometheus Pod scrape settings by adding the `insecure_skip_verify` flag to handle the self-signed certificate. For further details, refer to the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#tls_config).
 {{</call-out>}}
 
+## Installing Prometheus and Grafana
+
+### Prometheus
+
+```shell
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+helm install prometheus prometheus-community/prometheus -n monitoring --create-namespace --set server.global.scrape_interval=15s
+```
+
+Once running, you can access the Prometheus dashboard by using port-forwarding:
+
+```shell
+kubectl -n monitoring port-forward svc/prometheus-server 9090:80
+```
+
+Visit [http://127.0.0.1:9090](http://127.0.0.1:9090) to view the dashboard.
+
+### Grafana
+
+
+```shell
+helm repo add grafana https://grafana.github.io/helm-charts
+helm repo update
+helm install grafana grafana/grafana -n monitoring --create-namespace
+```
+
+Once running, you can access the Grafana dashboard by using port-forwarding:
+
+```shell
+kubectl -n monitoring port-forward svc/grafana 3000:80
+```
+
+Visit [http://127.0.0.1:3000](http://127.0.0.1:3000) to view the Grafana UI.
+
+#### Configuring Grafana
+
+In the Grafana UI menu, go to `Connections` then `Data sources`. Add your Prometheus service (`http://prometheus-server.monitoring.svc`) as a data source.
+
+Download the following sample dashboard and Import as a new Dashboard in the Grafana UI.
+
+{{< fa "download" >}} {{< link "static/grafana-dashboard.json" "`ngf-grafana-dashboard.json`" >}}
```

Comment:
This isn't quite working. Not sure what the exact syntax/location should be.

Reply:
Removing "static" from the path should do it. The shortcode is using the `relurl`, which is the path relative to the base path in the Hugo config. Hugo doesn't include the static directory in the path when it builds the site, so the url for the file in `static/grafana-dashboard.json` would be `<baseurl>/grafana-dashboard.json`.

Reply:
Thanks Jodie. That opened the json file properly in another tab, but I expected it to download automatically to my local machine.

Reply:
I need to make an update to the shortcode to tell it to download the file. This code won't have to change to support that behavior, so this is ok to commit as-is. I will make the change and give you a test theme version of the theme to try out.

Reply:
Actually, on second thought, I need to add a new shortcode for this. Changing how the link shortcode works to default to downloading could break stuff elsewhere.

Reply:
It use to behave by downloading. I know we have other docs that work like this, though I'm sure they are out of date with the latest stuff.

Reply:
@sjberman I created a new alpha release in the nginx-hugo-theme repo that you can use for testing:  https://github.com/nginxinc/nginx-hugo-theme/releases/tag/v0.41.0-alpha-1 

I updated your feature branch settings in netlify to use the alpha version already; if you want to use it locally, you'll need to run `hugo mod get https://github.com/nginxinc/nginx-hugo-theme v0.41.0-alpha-1` from the `site` directory.

If this works out ok, we'll merge it in and roll out a theme update for it. 

The syntax to use for the new shortcode is:

```
{{< download "static/grafana-dashboard.json" "grafana-dashboard.json" >}}
```

(I included the download icon in the shortcode, so you don't need to use the `fa` shortcode with this one.

Reply:
@jputrino Had some issues updating hugo locally, but I pushed the change. In the Netlify preview, the font is a bit small for the link, and it just opens a blank New Tab when clicking.

Reply:
@sjberman What browser are you using? DId you check to see if the file also downloaded? 
It worked fine for me in Safari, but I haven't tested other browsers yet.

Reply:
Firefox, wasn't downloaded.

Reply:
Apparently doesn't work on Chrome either.

Reply:
Thanks for that! We'll have to take a look and see what's up there. I don't want this to be a blocker in the meantime, so I'd say you can either stick with the `file` shortcode, provide a link directly to the file in Github, or create a Gist from it and embed that in the doc. Hugo has built-in support for gists: https://gohugo.io/content-management/shortcodes/#gist

Reply:
What does the `file` shortcode do?